### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by mpentek**
To be able to use the python_solvers_wrapper a file was added to have this for ALE as well. 
The ALEFractionalStep files were added (OpenMP tested, MPI not).

### CODE REVIEW

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        if (solver_type == "ALEMonolithic"):
+            solver_module_name = "ale_trilinos_navier_stokes_solver_vmsmonolithic"
+
+        elif (solver_type == "ALEFractionalStep"):
+            solver_module_name = "ale_trilinos_navier_stokes_solver_fractionalstep"
+
+        else:
+            raise Exception("the requested solver type is not in the python solvers wrapper")
+    else:
+        raise Exception("parallelism is neither OpenMP nor MPI")
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:40]: same

File: applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        super(ALETrilinosNavierStokesSolverFractionalStep, self).AddDofs()
+        # add ale dofs
+        self.ale_solver.AddDofs()
+        if mpi.rank == 0:
+            print("::[ALETrilinosNavierStokesSolverFractionalStep]:: DOFs ADDED.")
+
+    def Initialize(self):
+        super(ALETrilinosNavierStokesSolverFractionalStep, self).Initialize()
+        self.ale_solver.Initialize()
+                if mpi.rank == 0:
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py:52]: This appears wrongly formatted, might be in Git only

**Review by philbucher — CHANGES_REQUESTED** [review]

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    if (type(custom_settings) != KratosMultiphysics.Parameters):
+        raise Exception("input is expected to be provided as a Kratos Parameters object")
+
+    parallelism = custom_settings["problem_data"]["parallel_type"].GetString()
+    solver_type = custom_settings["solver_settings"]["solver_type"].GetString()
+
+    # Solvers for OpenMP parallelism
+    if (parallelism == "OpenMP"):
+
+        if (solver_type == "ALEMonolithic"):
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:19]: We recently had the discussion regarding the naming in the same File in the StructralMechApp and decided to follow the [Style Guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide#json-configuration-file)
[See here for example](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/python_solvers_wrapper_structural.py#L31)
Thus the name should be `ale_monolithic`

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            solver_module_name = "ale_navier_stokes_solver_vmsmonolithic"
+
+        elif (solver_type == "ALEFractionalStep"):
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:22]: `ale_fractionalstep`

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            solver_module_name = "ale_navier_stokes_solver_fractionalstep"
+
+        else:
+            raise Exception("the requested solver type is not in the python solvers wrapper")
+
+    # Solvers for MPI parallelism
+    elif (parallelism == "MPI"):
+
+        if (solver_type == "ALEMonolithic"):
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:31]: see above

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            solver_module_name = "ale_trilinos_navier_stokes_solver_vmsmonolithic"
+
+        elif (solver_type == "ALEFractionalStep"):
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:34]: see above

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    if (parallelism == "OpenMP"):
+
+        if (solver_type == "ALEMonolithic"):
+            solver_module_name = "ale_navier_stokes_solver_vmsmonolithic"
+
+        elif (solver_type == "ALEFractionalStep"):
+            solver_module_name = "ale_navier_stokes_solver_fractionalstep"
+
+        else:
+            raise Exception("the requested solver type is not in the python solvers wrapper")
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:26]: Here I suggest providing a bit more info, [see here for an example](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/python_solvers_wrapper_structural.py#L35)

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    elif (parallelism == "MPI"):
+
+        if (solver_type == "ALEMonolithic"):
+            solver_module_name = "ale_trilinos_navier_stokes_solver_vmsmonolithic"
+
+        elif (solver_type == "ALEFractionalStep"):
+            solver_module_name = "ale_trilinos_navier_stokes_solver_fractionalstep"
+
+        else:
+            raise Exception("the requested solver type is not in the python solvers wrapper")
```
Comment (philbucher) [inline applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:38]: More info, see above

**Review by philbucher — COMMENTED** [review]
Somehow my other comments got lost

File: applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py (hunk 2 above)
Comment (mpentek) [reply applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py:52]: Yes, correct, I have noted it, it is wrongly formatted. There was also one more bug in the Trilinos version. The problem is, I had no possibility to test it (which is obviously bad...).

File: applications/ALEapplication/python_scripts/ale_navier_stokes_solver_fractionalstep.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def CreateSolver(model_part, custom_settings):
+    return ALENavierStokesSolverFractionalStep(model_part, custom_settings)
+
+class ALENavierStokesSolverFractionalStep(navier_stokes_solver_fractionalstep.NavierStokesSolver_FractionalStep):
+
+    def __init__(self, model_part, custom_settings):
+        # remove the ale_settings so we can use the navier stokes constructor
+        print(custom_settings)
```
Comment (mpentek) [inline applications/ALEapplication/python_scripts/ale_navier_stokes_solver_fractionalstep.py:17]: _print(custom_settings)_ needs to be removed.

File: applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    def GetFluidSolver(self):
+        return super(ALETrilinosNavierStokesSolverFractionalStep, self)
+
+    def GetMeshMotionSolver(self):
+        return self.ale_solver
+
+    def Solve(self):
+        self.GetMeshMotionSolver().Solve()
+        self.GetFluidSolver.Solve()
```
Comment (mpentek) [inline applications/ALEapplication/python_scripts/ale_trilinos_navier_stokes_solver_fractionalstep.py:63]: _self.GetFluidSolver.Solve()_ is wrong, needs:  _self.GetFluidSolver().Solve()_

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py (hunk 1 above)
Comment (mpentek) [reply applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:40]: Here I do not seem to find an error when viewing in VS Code. Which line exactly or what should be formatted differently?

**Review by mpentek — COMMENTED** [review]
@philbucher The suggestions regarding naming: as ALE strongly depends on FLuidDynamics, I was sticking to the same naming convention, see applications/FluidDynamicsApplication/python_scripts/python_solvers_wrapper_fluid.py. I am totally ok with the change suggested (which then should be consistent in both), but I would first have the opinion of those more involved in the 2 applications.

File: applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py (hunk 1 above)
Comment (philbucher) [reply applications/ALEapplication/python_scripts/python_solvers_wrapper_ale.py:40]: Sorry I was referring to the other comment, which somehow got lost in the first review
I meant that also here you could provide more info regarding the available types

**Comment by philbucher** [comment]
The naming convention is a good point, lets hear what @KratosMultiphysics/fluid-dynamics thinks:
Guys, for the `solver_type`  in the `ALEApplication`, shall we stick to the Style Guide
 (=> `ale_monolithic` & `ale_fractionalstep`) or follow what is done in FluidDynamics (=> `ALEMonolithic` & `ALEFractionalStep`)?

**Comment by jcotela** [comment]
@philbucher We'd rather not change the naming convention in the FluidDynamics so close to the release. Having said that, we can discuss about unifying the solver names for the next release. If you want to go ahead with underscores, please use `fractional_step` and not `fractionalstep` where appropriate.

**Comment by philbucher** [comment]
hi @jcotela I was only talking abt ALE, ofc what you say is true.

**Comment by msandre** [comment]
Hey Mate, thanks for adding this. Please add a test example in the tests directory similar to the rectangle_2D3N_test but for the fractional step (you may need to rename the current test to monolithic). This will make clear if there are any more formatting problems with the script and you can use it in the future to detect possible changes affecting your solvers.

**Comment by mpentek** [comment]
@philbucher @jcotela I will use ale_monolithic and ale_fractional_step for ALE and keep those in FluidDynamics as are for now.

@msandre I will also add an example, first will probably update the pythons scrpits though

I have also remedied other bugs so will shortly do the commit.

**Comment by mpentek** [comment]
Closed, most actual status with the requested and recommended changes in #995.

**Pull request closed without merging**